<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>ODAP Overview</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="ODAP_manual_files/libs/clipboard/clipboard.min.js"></script>
<script src="ODAP_manual_files/libs/quarto-html/quarto.js"></script>
<script src="ODAP_manual_files/libs/quarto-html/popper.min.js"></script>
<script src="ODAP_manual_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="ODAP_manual_files/libs/quarto-html/anchor.min.js"></script>
<link href="ODAP_manual_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="ODAP_manual_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="ODAP_manual_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="ODAP_manual_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="ODAP_manual_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="ODAP_manual_files/libs/quarto-diagram/mermaid.min.js"></script>
<script src="ODAP_manual_files/libs/quarto-diagram/mermaid-init.js"></script>
<link href="ODAP_manual_files/libs/quarto-diagram/mermaid.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">ODAP Overview</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<!--
quarto render ODAP_manual.qmd
-->
<section id="odap-overview" class="level1">
<h1>ODAP Overview</h1>
<blockquote class="blockquote">
<p><em>This is a shared description for the internal team of the core purpose, scope, operational components, and governance structure of the ODAP. It provides an introduction, plan and a reference.</em></p>
</blockquote>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>ODAP is long term, UK wide, 4 nations outbreak response infrastructure providing the unique and scalable capabilities of the academic sector to tackle public health challenges. ODAP creates a UK-wide capability by curating and linking outbreak relevant data from clinical records, research studies and audit data. It brings together key initiatives and leadership across the UK including ISARIC, COG-UK, MRC CLIMB and GenOMICC. The platform combines a national Trusted Research Environment (TRE) infrastructure collocated with &gt;£100M of world-class computational and data science capacity including the UK National Supercomputer, with a UK-wide governance framework.</p>
</section>
<section id="what-is-the-outbreak-data-analysis-platform-odap" class="level2">
<h2 class="anchored" data-anchor-id="what-is-the-outbreak-data-analysis-platform-odap">What is the Outbreak Data Analysis Platform (ODAP)?</h2>
<p>The ODAP is the overarching term for a range of computers supporting research within a specific scope. These include the ODAP TRE (formerly called: flexible compute space; ultra; ultra2), and various project spaces in the National Safe Haven.</p>
<p>It has two components:</p>
<ol type="1">
<li>The ODAP Core. <em>Getting fast answers to important public health research questions.</em> ODAP Core is a complicated network of overlapping, linked data from research studies and national datasets, created through a range of independent research activities, sitting in multiple computer systems. These datasets are linked through the ISARIC Clinical Characterisation Protocol (CCP).
<ul>
<li>Purpose: to prepare for and support urgent public health research.</li>
<li>Access: data are openly accessible to appropriately-qualified researchers studying questions within the scope of the platform by individual agreement with the data providers. Much of the data is under embargo and publication of results requires consent from the data providers. Put simply, access is open but not easy.</li>
<li>Funding: The core ODAP is funded by the Baillie Gifford Pandemic Science Hub.</li>
</ul></li>
<li>ODAP Open - <em>Making data access easy for approved researchers.</em> ODAP Open will be a robust, fully-staffed data access governance system to lower the barrier to entry by taking over data controllership from data contributors, so that a single approval is required for data access.
<ul>
<li>Funding: not in place. Funding from HDR UK infrastructure team is anticipated.</li>
<li>Leadership: open call for a senior appointment when funding is available.</li>
<li>Access: data are openly accessible to appropriately-qualified researchers studying questions within the scope of the platform following a single, streamlined application</li>
</ul></li>
</ol>
<p>The foundation for the ODAP is the International Severe Acute Respiratory Infection Consortium (ISARIC) Clinical Characterisation Protocol (CCP).</p>
</section>
</section>
<section id="odap-core" class="level1">
<h1>ODAP Core</h1>
<section id="scope-of-the-odap" class="level2">
<h2 class="anchored" data-anchor-id="scope-of-the-odap">Scope of the ODAP</h2>
<p><strong>The purpose of the ODAP is to facilitate biomedical research to advance understanding of severe infectious disease* and other exposures of public health interest.** Research within the ODAP is strictly limited to this purpose.</strong></p>
<p>* Severe infectious disease - this term describes all severe infectious agents, including new, re-emerging or therapy-resistant forms of existing infectious agents.</p>
<p>** Other exposures of public health interest: this term describes new or unexplained poisoning, or exposure to harmful energy sources such as electromagnetic radiation.</p>
<p>Examples of in-scope research: - Understanding the evolution and biology of SARS-CoV-2. - Understanding the epidemiology and transmission of SARS-CoV-2. - Understanding COVID-19 disease risk, severity and outcomes. - Monitoring and understanding the impact of non-pharmacological interventions against SARS-CoV-2 transmissions and COVID-19 disease. - Monitoring, understanding, and assessing the impact of treatments, vaccines and prior infections in COVID-19 disease. - Analysing or modelling SARS-CoV-2 and COVID-19 data for future pandemic preparedness.</p>
<p>Examples of out of scope activities - Their research questions primarily focusses on a non-infectious disease area with incidental involvement of infectious disease (Focussed on a pathogen with no potential public health concern)</p>
</section>
<section id="isaric-clinical-characterisation-protocol-ccp" class="level2">
<h2 class="anchored" data-anchor-id="isaric-clinical-characterisation-protocol-ccp">ISARIC Clinical Characterisation Protocol (CCP)</h2>
<p>The scope of the ODAP mirrors the objective of the CCP, an ethically-approved research study in the UK (Joint Chief Investigators: Calum Semple(Liverpool, Oxford) and Kenneth Baillie (Edinburgh, Oxford)). A broad range of scientists with relevant expertise have come together to form a UK-wide group: the ISARIC Comprehensive Clinical Characterisation Collaboration. Membership of this collaboration is by invitation and is extended to researchers performing high-quality biomedical research to advance understanding of severe infectious disease and other exposures of public health interest.</p>
<section id="the-isaric-spine" class="level3">
<h3 class="anchored" data-anchor-id="the-isaric-spine">The ISARIC Spine</h3>
<p>Some studies, addressing aims that are within the scope of the CCP, have chosen to contribute data into the ODAP for linkage to CCP and other datasets. Because these studies are addressing questions that are directly within the scope of the CCP, where appropriate, participants in these studies are eligible to be enrolled under the CCP.</p>
<p>Under our existing protocol and approvals for the ISARIC CCP, we have recruited over 300,000 Covid patients without explicit consent, into an observational study, with clear and specific aims. We are now linking this data to other studies, including consented studies such as GenOMICC and RECOVERY, and non-consented studies such as COG-UK viral sequencing.</p>
<p>Applying the established governance and data handling procedures for the ISARIC CCP to the additional linked data from these studies will enable cross-cutting research, all within the specific aims of the CCP. To achieve this, we have agreed to recruit all patients in the following studies into the ISARIC CCP, if they are known to meet the inclusion criteria for the CCP:</p>
<ul>
<li>GenOMICC</li>
<li>RECOVERY</li>
<li>PHOSP</li>
<li>COG-UK</li>
<li>HEAL-COVID</li>
</ul>
<p>In each case, we will only recruit patients who meet the inclusion criteria for CCP. In our view, there is no ethical or information governance difference between recruiting a patient remotely from a hospital, or recruiting them remotely from within our trusted research environment. This has been agreed with the ISARIC CCP study sponsor and is explicitly stated in CAG and PBPP applications.</p>
<section id="examples" class="level4">
<h4 class="anchored" data-anchor-id="examples">Examples</h4>
<p>The ISARIC4C consortium will incorporate participants from other studies into ISARIC4C study, creating a superset of participants in a single “spine”. Doing so would allow them to match to NHS data under existing ISARIC4C data agreements.</p>
<p>To do this, we need to have an ID and an NHS number, or CHI number, as a minimum standard. Some participants in the other studies may already be in ISARIC4C. These need to be identified and tagged as being the same individual. This will be done by comparison of NHS or CHI numbers. The spine will contain all the joins between projects and participants. Those participants that are not already in the ISARIC4C system will have an ISARIC4C ID allocated to them.</p>
<p>The data from the spine will be used for several purposes:</p>
<ul>
<li><p>it will feed into the NHS Digital data matching system. This system requires ID, NHS number, date of birth and postcode (the latter two are optional). This will allow the ongoing monthly matching of study participants to English &amp; Welsh Health Data for the superset of ISARIC4C participants.</p></li>
<li><p>it will allow study participants to be identified against a cohort of NHS or CHI numbers for individual research studies, and as requested by Trusted Research Environments (TREs).</p></li>
</ul>
<p>Currently the data for participants in these studies are held in a series of REDCap databases. These are the source of truth for the data and update on a regular basis. We will therefore need a process to update the spine from these sources.</p>
</section>
</section>
</section>
<section id="data-access" class="level2">
<h2 class="anchored" data-anchor-id="data-access">Data access</h2>
<p>Some data providers choose to delegate approval of data access to the ODAP team. The diagram in <a href="#fig-access">Figure&nbsp;1</a> describes the five safes application form and light-touch review process. These processes are overseen by the Data Access Governance Committee (DAGC).</p>
<div id="fig-access" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><embed src="images/data_access/Data_access_flow_v0.8.pdf" class="img-fluid"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Data access flows</figcaption><p></p>
</figure>
</div>
<p>The data access review processes are described in detail in the following documents:</p>
<ul>
<li><a href="ODAP_open/DataAccessReviewProcess.md">Data Access Review Process</a></li>
<li><a href="ODAP_open/DAGC_ToR.md">DAGC Terms of Reference</a></li>
<li><a href="ODAP_open/PRP_ToR.md">PRP Terms of Reference</a></li>
<li><a href="ODAP_open/PSG_ToR.md">PSG Termos of Reference</a></li>
</ul>
</section>
<section id="computer-architecture" class="level2">
<h2 class="anchored" data-anchor-id="computer-architecture">Computer architecture</h2>
<p>The ODAP consists of two main compute areas.</p>
<section id="flexible-compute-space" class="level3">
<h3 class="anchored" data-anchor-id="flexible-compute-space">Flexible Compute Space</h3>
<p>The Flexible Compute Space (FCS) is a Trusted Research Environment with large scale compute capacity and access to a range of software tools for data analysis and machine learning. Data within the FCS can be either under embargo, or not under embargo.</p>
<p>Access to the FCS is provided to researchers through <em>private project zones</em></p>
</section>
<section id="phs-national-safe-haven" class="level3">
<h3 class="anchored" data-anchor-id="phs-national-safe-haven">PHS National Safe Haven</h3>
<p>Some data remains within the PHS National Safe Haven but will transfer to the FCS on completion of the Systems Security Policy.</p>
</section>
</section>
<section id="data-ingress" class="level2">
<h2 class="anchored" data-anchor-id="data-ingress">Data ingress</h2>
<p>Data are linked into the ISARIC Spine using the patient’s NHS number by PHS before transfer to the FCS as shown in <strong>?@fig-flow</strong>.</p>
<div class="cell" data-fig-width="10">
<div class="cell-output-display">
<div>
<p>
</p><pre class="mermaid mermaid-js" data-tooltip-selector="#mermaid-tooltip-1">graph LR

New(New Dataset) --&gt; ISARIC3

subgraph ODAP
  subgraph Flexible Compute Space
    ISARIC4C[ISARIC Spine] --- NHS[NHS England]  
    ISARIC4C --- GenOMICC  
    ISARIC4C --- PHOSP     
    ISARIC4C --- COG-UK  
    ISARIC4C --- SICSAG  
    Derived[Derived&lt;br&gt;variables]
  end
  ISARIC3 ==&gt; to_fcs[(IDs allocated&lt;br&gt;Transfer to FCS)]
  to_fcs ==&gt; ISARIC4C
  subgraph Safe Haven
    NHS_IDS[\Master link&lt;br&gt;to NHS numbers/] --&gt;|PHS team only| ISARIC3((ISARIC Spine&lt;br&gt;ID allocation))
    RECOVERY --&gt; ISARIC2[ISARIC-CCP&lt;br&gt;extract]
  end
end

Derived ---&gt;|API|PHA([Public&lt;br&gt;Health&lt;br&gt;Agencies])
Derived ---&gt;|API|TRE([External&lt;br&gt;TREs])

</pre>
<div id="mermaid-tooltip-1" class="mermaidTooltip">

</div>
<p></p>
</div>
</div>
</div>
<p>{fig-mermaid}</p>
</section>
<section id="api" class="level2">
<h2 class="anchored" data-anchor-id="api">API</h2>
<p>There will be a secure API in place from the Flexible Compute Space to External TREs and UK Public Health Agencies. This is to accelerate research by enabling data linkage across multiple data sets using unique identifiers to facilitate secure transfer of specified data fields e.g., viral sequencing data. The future purpose of the API is more in line with providing a mechanism for quick pandemic updates to national health services.</p>
</section>
<section id="data-contributors" class="level2">
<h2 class="anchored" data-anchor-id="data-contributors">Data Contributors</h2>
<p>Each data contributor will explicitly consent to the lifting of any embargo on their data. A permanent record of these instructions will be held in a shared file. <!--FILE PATH TO MASTER RECORD OF EMBARGO STATUS IN SECURE LOCAION XXX --></p>
</section>
<section id="dataset-catalogue" class="level2">
<h2 class="anchored" data-anchor-id="dataset-catalogue">Dataset catalogue</h2>
<p>The ODAP dataset catalogue is a flexible central record of the datasets held within the platform, approval requirements for access, embargo status, and contents. Each row of the catalogue provides information about an <code>atomic dataset</code>.</p>
<p>An <code>atomic dataset</code> should be the smallest unit of data that a user would apply for access to. It will have a version number, source file path(s), and named data controllers and data processors. Any legal agreement that applies to part of an <code>atomic dataset</code> will apply to all of the data within it. Any user who has access to part of an <code>atomic dataset</code> has access to all of the data within it. Where possible an <code>atomic dataset</code> will be in tidy format.</p>
<p>A <code>compound dataset</code> is a group of <code>atomic datasets</code> that commonly go together. They are grouped together for convenience because researchers often need access to all of them at once.</p>
<p>A live <a href="https://uoe.sharepoint.com/:x:/r/sites/ISARIC4C/DataInfrastructureAndGovernance/ODAP/_documents/ODAP%20datasets%20mapping.xlsx?d=w1dde5dbd6a094799b1595de102fa3818&amp;csf=1&amp;web=1&amp;e=coUYiB">ODAP datasets mapping.xlsx</a> is available for reference. [Further details about processes and approvals to be added]</p>
<p>A parallel document exists for <a href="https://uoe.sharepoint.com/:x:/r/sites/ISARIC4C/DataInfrastructureAndGovernance/Research%20Projects/ISARIC%20eDRIS%20Projects.xlsx?d=w8624bebadf7f491abc6add8587a039d8&amp;csf=1&amp;web=1&amp;e=npUB8k">eDRIS</a></p>
<p>Applications and approved users are found here: https://uoe.sharepoint.com/:x:/r/sites/ISARIC4C/DataInfrastructureAndGovernance/IDAMAC/IDAMAC_Requests_Pipeline.xlsx?d=w7431f37efd1e4f53acfb2753fc03104f&amp;csf=1&amp;web=1&amp;e=S0Bit5</p>
<p>Dataset lists and embargo status is recorded here: XXX</p>
</section>
</section>
<section id="opap-open" class="level1">
<h1>OPAP Open</h1>
<p>Once the embargo is lifted by a data contributor, data provision will be <em>FAIR</em> (Findable, Accessible, Interoperable, Reusable) and access will adhere to the <em>five safes</em> principles:</p>
<ul>
<li>Safe data: data is treated to protect any confidentiality concerns.</li>
<li>Safe projects: research projects are approved by data owners for the public good.</li>
<li>Safe people: researchers are trained and authorised to use data safely.</li>
<li>Safe settings: a SecureLab environment prevents unauthorised use.</li>
<li>Safe outputs: screened and approved outputs that are non-disclosive.</li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>